ANGULAR/Task_2/User-Management-App/src/app/shared/user-details/user-details.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+import { UsersService } from 'src/app/users.service';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-user-details',
+  templateUrl: './user-details.component.html',
+  styleUrls: ['./user-details.component.css']
+})
+export class UserDetailsComponent implements OnInit {
+
+  paramsId:any;
+  user:any;
+  constructor(private userService:UsersService, private route:ActivatedRoute,) { }
+
+  ngOnInit() {
+    this.paramsId = +this.route.snapshot.params['id'];
```

Comment:
try to put a negative condition here and see if it works. If the params does not have a field ID then it will fail. Assign a default value to this 


this.paramsId = +this.route.snapshot.params['id'] || 0;